sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/CountingSource.java

```diff
@@ -302,17 +356,27 @@ public void validate() {}
    */
   private static class UnboundedCountingReader extends UnboundedReader<Long> {
     private UnboundedCountingSource source;
+
     private long current;
     private Instant currentTimestamp;
 
+    private Instant started;
+    private long totalProduced;
+
     public UnboundedCountingReader(UnboundedCountingSource source, CounterMark mark) {
       this.source = source;
       if (mark == null) {
         // Because we have not emitted an element yet, and start() calls advance, we need to
         // "un-advance" so that start() produces the correct output.
         this.current = source.start - source.stride;
+        // Do not produce an element immediately, to ensure that we do not exceed the requested
+        // period due to splits.
```

Comment:
rm'd the comment; moved to start (have to add a null check to make sure we don't overwrite our checkpoint)